src/pages/Wallet/SendPage.tsx

```diff
@@ -97,32 +101,83 @@ const SendPage = () => {
         )
 
         const { txId, unsignedTx } = txCreateResp.data
+        const txSendResp = await signAndSendTransaction(txId, unsignedTx, address, wallet.privateKey)
+
+        if (txSendResp) {
+          addPendingTx({
+            txId: txSendResp.data.txId,
+            toAddress: address,
+            timestamp: new Date().getTime(),
+            amount: fullAmount,
+            type: 'transfer'
+          })
+        }
 
-        const signature = client.clique.transactionSign(txId, wallet.privateKey)
+        setSnackbarMessage({ text: 'Transaction sent!', type: 'success' })
+        history.push('/wallet')
+      } catch (e) {
+        // TODO: When API error codes are available, replace this substring check with a proper error code check
+        if (isHTTPError(e) && e.error?.detail && e.error.detail.includes('consolidating')) {
+          setIsConsolidateUTXOsModalVisible(true)
+        } else {
+          setSnackbarMessage({
+            text: getHumanReadableError(e, 'Error while sending the transaction'),
+            type: 'alert',
+            duration: 5000
+          })
+        }
+      }
 
-        const txSendResp = await client.clique.transactionSend(wallet.address, unsignedTx, signature)
+      setIsSending(false)
+    }
+  }
 
-        addPendingTx({
-          txId: txSendResp.data.txId,
-          toAddress: address,
-          timestamp: new Date().getTime(),
-          amount: fullAmount
-        })
+  const consolidateUTXOs = async () => {
+    if (client && wallet) {
+      setIsConsolidating(true)
+      try {
+        const txCreateResp = await client.clique.transactionConsolidateUTXOs(
+          wallet.publicKey,
+          wallet.address,
+          wallet.address
+        )
+        for (const { txId, unsignedTx } of txCreateResp.data.unsignedTxs) {
+          const txSendResp = await signAndSendTransaction(txId, unsignedTx, wallet.address, wallet.privateKey)
+          if (txSendResp) {
+            addPendingTx({
+              txId: txSendResp.data.txId,
+              toAddress: wallet.address,
+              timestamp: new Date().getTime(),
+              amount: '',
+              type: 'consolidation'
```

Comment:
It's better to define `'consolidation'` as constants, so it's easier to modify in the future, or to avoid typos

Reply:
done

Reply:
@polarker actually, TypeScript does not allow us to do typos:

![image](https://user-images.githubusercontent.com/1579899/149305202-39d06bdc-e366-41d5-b9a3-6cc418580988.png)

As for easier to modify in the future, we can easily do a global search+replace of the strings.